src/foam/core/Action.js

```diff
@@ -157,20 +171,22 @@ foam.CLASS({
     },
 
     function createIsAvailable$(data$) {
-      return this.createSlot_(this.isAvailable, data$);
+      return this.createSlot_(this.isAvailable, 'available', data$);
     },
 
-    function maybeCall(ctx, data) {
-      // TODO: permission check
-      if ( this.isEnabledFor(data) && this.isAvailableFor(data) ) {
+    async function maybeCall(ctx, data) {
```

Comment:
This is going to cause problems if an Action ever does window.open() to say open a link in another window it will run up against popup blockers which only let you open windows during synchronous handling of a 'click' event handler.

Not a game breaking restriction but something to keep in mind.